components/contact-us-form.tsx

```diff
@@ -0,0 +1,193 @@
+        //         ),
+        //     });
+        //     alert("Message sent successfully!");
+        // } catch (error) {
+        //     console.error("Error sending email:", error);
+        //     alert("Failed to send message. Please try again later.");
+        // }
+    };
+
+    return (
+
+        <div className="flex flex-col md:flex-row justify-between p-6 text-white">
+
+            <div className="md:w-1/2 mb-6 flex flex-col items-center">
+
+                <h2 className="animate-[gradient_6s_linear_infinite] bg-[linear-gradient(to_right,theme(colors.gray.200),theme(colors.forest.200),theme(colors.gray.50),theme(colors.forest.300),theme(colors.gray.200))] bg-[length:200%_auto] bg-clip-text pb-4 font-menda text-4xl font-medium text-transparent md:text-5xl text-center">
+
+                    <span className="block">Let&apos;s</span>
+
+                    <span className="block">Connect</span>
+
+                </h2>
+
+                <div className="flex flex-col mt-4 content-center">
+
+                    <div className="flex items-center">
+
+                        <img src="/images/icons/mail-outline.svg" alt="Email" className="mr-4" style={{ width: '1rem', height: '1rem' }} />
+
+                        <span>marketing@luminate.com</span>
+
+                    </div>
+
+                    <div className="flex items-center">
+
+                        <img src="/images/icons/phone-flip.svg" alt="Email" className="mr-4" style={{ width: '1rem', height: '1rem' }} />
+
+                        <span>+92 3361000001</span>
+
+                    </div>
+
+                    <div className="flex items-center">
+
+                        <img src="/images/icons/location-pin.svg" alt="Email" className="mr-4" style={{ width: '1rem', height: '1rem' }} />
+
+                        <span>Islamabad, Pakistan</span>
+
+                    </div>
+
+                </div>
+
+            </div>
+
+            <div className="md:w-1/2">
+
+                <h2 className="animate-[gradient_6s_linear_infinite] bg-[linear-gradient(to_right,theme(colors.gray.200),theme(colors.forest.200),theme(colors.gray.50),theme(colors.forest.300),theme(colors.gray.200))] bg-[length:200%_auto] bg-clip-text pb-4 font-menda text-3xl font-medium text-transparent md:text-3.5xl">
+
+                    Let&apos;s Collaborate to Create Something Extraordinary
+
+                </h2>
+
+                <form onSubmit={handleSubmit(onSubmit)} style={{ display: 'flex', flexDirection: 'column' }}>
+                    <div style={{ marginBottom: '15px' }}>
+                        <label style={{ color: 'white', marginBottom: '5px' }}>Name</label>
+                        <input
+                            type="text"
+                            placeholder="Name"
+                            {...register("name")}
+                            style={{
+                                width: '100%',
+                                padding: '10px',

```

Comment:
Please refactor this form component into another component which will be client side rendered, and handle the functionality of the of the form itself like sending emails etc. 

this correlates directly to the encapsulation concept in OOP

Reply:
Also make this component server side rendered.